src/cockpit/channels/filesystem.py

```diff
@@ -218,7 +267,7 @@ async def set_contents(
                         raise ChannelError('change-conflict')
                     # chown/chmod from the existing file permissions
                     os.fchmod(fd, stat.S_IMODE(buf.st_mode))
-                    os.fchown(fd, buf.st_uid, buf.st_gid)
+                    chown_if_required(fd, buf)
```

Comment:
I'm not sure I agree with this..... what's the usecase for replacing a file and changing the ownership when doing so, but keeping the old permissions?

How might this interact with set[ug]id flags?

Reply:
Actually not sure what you mean here? You mean setting only mode but not changing the user/group?

Reply:
Right, I agree that the attrs should win over tag as it is more specific? And by reading this is what we do now?